### ISSUE TITLE
how to compile mxnet with cpp and gpu support in the docker ?

### ISSUE THREAD

**Issue by MyraBaba** [apache/incubator-mxnet#17216]
Hi,

I asked in the forum but non answer received

I couldn't find a docker image that has gpu and cpp support. 

I want to compile myself in the docker.

is it possible ?

how can I do that ant guidance?

**Comment by QueensGambit** [comment]
Hi @MyraBaba ,
there is an official MXNet NVIDIA docker image available which also has TensorRT support:
* https://docs.nvidia.com/deeplearning/frameworks/mxnet-release-notes/rel_19-12.html#rel_19-12

I also created an examplary Dockerfile on how to use it:
* https://github.com/QueensGambit/CrazyAra/blob/master/engine/src/rl/Dockerfile

**Comment by MyraBaba** [comment]
So is this include mxnet cpp support and all required include files for cpp mxnet ?

**Comment by QueensGambit** [comment]
Yes, the docker file contains a shared object library at:
* /usr/local/lib/libmxnet.so

which was built with C++ and TensorRT support.

---

The cpp include files are located at:
* /opt/mxnet/cpp-package/include

within the docker container.

**Comment by MyraBaba** [comment]
the docker file could create the image.

After awhile it says : 

> unauthorized autjenticatin required

**Comment by MyraBaba** [comment]
also Ican compile cpp-mxnet/example/inference . with make

it gave : fatal error mxnet/op.h . No such file

**Comment by litaotju** [comment]
If you are trying to build mxnet from source, actually there are already docker file included in this repo. 

- ci/docker/Dockerfile.build.ubuntu_gpu_tensorrt
- ci/docker/Dockerfile.build.ubuntu_gpu_cu101

You could use anyone of these for your purpose. For example, if you need TensorRT and gpu support. Use the following one to make a docker:

```
 docker build -t mxnet-dev-trt/trt-515-cu-100 -f ./Dockerfile.build.ubuntu_gpu_tensorrt 
```

**Comment by QueensGambit** [comment]
> it gave : fatal error mxnet/op.h . No such file

Yes, I don't really like the procedure either to create a cpp-header file with a python script.
You need to call `OpWrapperGenerator.py` to create the `op.h` file:

```bash
cd /opt/mxnet/cpp-package/scripts
python OpWrapperGenerator.py "/usr/local/lib/libmxnet.so"
```

If this doesn't work on your system due to encoding errors, you can try downloading the file instead:

* https://github.com/dmlc/MXNet.cpp/blob/master/include/mxnet-cpp/op.h
